NossoMercadoLivre/Repositories/Interfaces/IUserRepository.cs

```diff
@@ -5,5 +5,6 @@ namespace NossoMercadoLivre.Repositories
     public interface IUserRepository : IRepository<User>
     {
         bool AnyUserByUsername(string username);
+        User GetUserByUsername(string username);
```

Comment:
Se não existir retorna nulo?

Reply:
Sim

Reply:
*Não retornamos nulo dentro das regras da aplicação. Pense que seu computador vai explodir.*

Essa é um pilar muito importante. Você está dizendo que retorna User, mas pode retornar nulo. Como que o outro ponto do código sabe disso? Aqui o sistema aumenta demais a chande de NPE. Minha sugestão é evitar isso deixando pistas. Neste caso a pista é para dizer que o retorno aqui pode ser nulo. No Java usamos o Optional e no C#, pelo que pesquisei, na versão 8 tem as Nullable References. https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references 

E aqui tem um belo exemplo https://dev.to/integerman/safer-code-with-c-8-non-null-reference-types-4f2c